### PR TITLE
added js3 week3 Homework

### PR DESCRIPTION
**Pull request by STEP0HEN**
Still working on the Shopping  Cart.

### CODE REVIEW

File: javascript/javascript3/week3/JS3-WEEK3-HomeWork/Classes-Canvas/main.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        ctx.fill();
+        ctx.stroke();
+    };
+};
+
+const circle1 = new Circle(150,90, 50, 0,2* Math.PI, 'black')
+console.log(circle1);
+circle1.draw();
+
+const c2 = new Circle(70, 70, 30, 0, 2 * Math.PI, 'turquoise');
```
Comment (nmoskaleva) [inline javascript/javascript3/week3/JS3-WEEK3-HomeWork/Classes-Canvas/main.js]: Don't use single-letter names such as `c1` for variables – it's bad for readability.

File: javascript/javascript3/week3/JS3-WEEK3-HomeWork/Promises-Githubrepo/main.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const render = (item) => {
+  const liTag = document.createElement("li");
+  liTag.textContent = (item);
+  bodyTag.appendChild(liTag)
+};
+
+const githubApi1 = `https://api.github.com/search/repositories?q=user:STEP0HEN`
+const githubApi2 = `https://api.github.com/search/repositories?q=user:AkoAyukngat`
+const githubApi3 = `https://api.github.com/search/repositories?q=user:DeepthiDukka`
```
Comment (nmoskaleva) [inline javascript/javascript3/week3/JS3-WEEK3-HomeWork/Promises-Githubrepo/main.js]: Imagine you would have many users, not just 3: creating a variable for each user wouldn't be an option. You could do 
`const url = 'https://api.github.com/search/repositories?q=user:'`
and add names to it.

File: javascript/javascript3/week3/JS3-WEEK3-HomeWork/Promises-Githubrepo/main.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const myClassmatesApi = () => {
+    return Promise.all(classmatesApi)
+    .then(response => {
+        const jsonResponse1 = response[0].json();
+        const jsonResponse2 = response[1].json();
+        const jsonResponse3 = response[2].json();
+
+        return Promise.all([jsonResponse1,jsonResponse2,jsonResponse3]);
+    });
+};
```
Comment (nmoskaleva) [inline javascript/javascript3/week3/JS3-WEEK3-HomeWork/Promises-Githubrepo/main.js:26]: Think how to make this code reusable. The best way would be to write a separate function to fetch user data. 
`return Promise.all(classmatesApi)` – you're just fetching data here, and you can fetch in a more simple way.

File: javascript/javascript3/week3/JS3-WEEK3-HomeWork/Shopping-cart-classes/main.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        this.products = [];
+    };
+    
+    addProduct(product) {
+        this.products.push(product)
+    };
+    
+    removeProduct(product) {
+        const  newProducts = this.products.filter((existingProduct => {
+            return existingProduct.name !== product.name && product.price
```
Comment (nmoskaleva) [inline javascript/javascript3/week3/JS3-WEEK3-HomeWork/Shopping-cart-classes/main.js]: why `product.price`?

**Review by nmoskaleva — COMMENTED** [review]
Well done! But try to make your code reusable in `promises-githubrepo`.

**Comment by STEP0HEN** [comment]
Thank you so much for the review. I've implemented the changes

**Pull request merged**